presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1479,4 +1480,44 @@ public ActualProperties getProperties()
             return properties;
         }
     }
+
+    private static boolean isPartitioningCompatible(List<PartitioningHandle> partitioningHandles)
```

Comment:
https://github.com/prestodb/presto/pull/13091#issuecomment-513345738

While working on this PR further, I also found out that I need to change SPI to define the relationship between two partitioning. I'm considering adding it to `PartitioningHandle` and `ConnectorPartitioningHandle`. It should not block what you are doing now but I guess it might get refactored, just so you know.